meta-psa/recipes-security/psa-arch-tests/psa-arch-tests_0.9.bb

```diff
@@ -0,0 +1,113 @@
+# Copyright (c) 2019 Arm Limited and Contributors. All rights reserved.
+#
+# SPDX-License-Identifier: MIT
+
+HOMEPAGE = "https://github.com/ARM-software/psa-arch-tests"
+
+DESCRIPTION = "\
+The Arm Platform Security Architecture Test Suite provides tests \
+for implementations of PSA specifications. \
```

Comment:
>  the recipe is not robust to internal project changes (e.g. dir renames).

Agree.

> It would be better to have a well defined interface e.g. the psa-arch-test Makefile has OPTION=VALUE settings that enable the required behaviour.

Agree

> This means upstreaming changes which i understand is problematic.

Yes, I'd like to upstream more changes than are in the patches here. Mainly to provide an option to build a `main()` function and link everything together. I think the current set of patches is less controversial to the code owners than that so I'm trying to get that in first.

> It would be helpful to highlight things which are currently (or could be in the future) part of a logical interface (e.g. SUITE, TARGET) and things that shouldn't be part of the interface (PSA_LIB_OPTIONS?, main.c, the code in compile_main_obj(), using setup.sh to build lib.o files).

I think that's going a bit far. Whatever I think a sensible interface for building psa-arch-tests is, it probably won't happen.

> mbed-crypto can be built natively for x86. Can psa-arch-test?

Yes, I've run the tests against Mbed Crypto on my x86_64 Ubuntu desktop.

> How would I build psa-arch-${suite}-tests for x86?

Do approximately what this recipe does:
1. Apply the patches from this PR to psa-arch-tests.
2. Install Mbed Crypto.
3. Set `CC`, `CFLAGS`, `LD`, `LDFLAGS`, `AS`, `ASFLAGS`, `AR`, and `ARFLAGS` appropriately for your target.
4. Build test libs:
    ```
    api-tests/tools/scripts/setup.sh \
        --source api-tests \
        --build /path/to/build/dir \
        --target  tgt_dev_apis_stdc \
        --archive_tests \
        --suite crypto \
        --toolchain ENV \
        --include /usr/local/include
    ```
   Replacing `/usr/local/include` with where Mbed Crypto left its header files.
5. Compile `main.c` from this recipe into an object.
6. Link everything together. At the end of `setup.sh`'s output it will list the `.a` files it created that you have to link together with your `main()` object. Specify `-lmbedcrypto`.
7. Run `/path/to/build/dir/psa-arch-crypto-tests`.